werkzeug/_reloader.py

```diff
@@ -107,9 +107,19 @@ def _walk(node, path):
 
 
 class ReloaderLoop(object):
-    _sleep = time.sleep  # monkeypatched by testsuite
+
     name = None
 
+    # monkeypatched by testsuite. this used to just say `_sleep = time.sleep`,
+    # but because c functions behave differently to python functions when
+    # attached to classes, this breaks if sleep has been already replaced with
+    # a python function (e.g. by eventlet.monkey_patch). python functions are
+    # descriptors and become methods when attahed to classes, whereas c
+    # functions remain as functions (thus behaving like staticmethods)
```

Comment:
it's not monkeypatching that fails, it's _using_ `ReloaderLoop._sleep` if `sleep` was monkeypatched already _before_ it was attached to the `ReloaderLoop`